sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -776,16 +938,55 @@ public HttpPipeline getHttpPipeline() {
         Function<String, Mono<PagedResponse<PeekedMessageItem>>> retriever =
             marker -> StorageImplUtils.applyOptionalTimeout(this.client.getMessages()
                 .peekWithResponseAsync(queueName, maxMessages, null, null, context), timeout)
-                .map(response -> new PagedResponseBase<>(response.getRequest(),
-                    response.getStatusCode(),
-                    response.getHeaders(),
-                    response.getValue(),
-                    null,
-                    response.getDeserializedHeaders()));
+                .flatMap(this::transformMessagesPeekResponse);
 
         return new PagedFlux<>(() -> retriever.apply(null), retriever);
     }
 
+    private Mono<PagedResponseBase<MessagesPeekHeaders, PeekedMessageItem>> transformMessagesPeekResponse(
```

Comment:
The logic for both paged response transformations is very similar, is it possible to create a more generic method which takes a few generic `Function`s or `Supplier`s and centralize the logic?

Reply:
It has reduced a bit after making it all reactive. so amount of similar code is less now. I'm afraid that generalizing this will produce something that's not easily understandable for poor soul debugging it.